File: tasks/implementations/python/tasks/file_io_large.py
from pathlib import Path

def run(size: str, fixtures_root: str) -> None:
    path = Path(fixtures_root) / "generated" / "file_io" / f"{size}.txt"
    total = 0
    with path.open("r", encoding="utf-8") as handle:
        for line in handle:
            total += int(line.strip())
    print(total)

File: tasks/implementations/rust/tasks/file_io_large.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

pub fn run(size: &str, fixtures_root: &str) {
    let path = format!("{}/generated/file_io/{}.txt", fixtures_root, size);
    let file = File::open(path).unwrap();
    let mut total = 0i64;
    for line in BufReader::new(file).lines() {
        total += line.unwrap().parse::<i64>().unwrap();
    }
    println!("{}", total);
}
